src/main/resources/react4xp/components/parts/MovieProcessor.ts

```diff
@@ -0,0 +1,101 @@
+import type { Content } from '@enonic-types/lib-content';
+import type { PartComponentProcessorFunction } from '@enonic-types/lib-react4xp/DataFetcher';
+import {imageUrl, pageUrl} from '/lib/xp/portal';
+import { get as getContentByKey } from '/lib/xp/content';
+
+function fetchAdditionalPhotos(photoIds: string[]) {
+    return photoIds.map(photoId => {
+        const photoContent = getContentByKey<Content>({ key: photoId });
+        return {
+            _id: photoContent._id,
+            title: photoContent.displayName,
+            imageUrl: imageUrl({ id: photoContent._id, scale: 'width(250)' }) // Image scaled for remaining photos
+        };
+    });
+}
+
+export const movieProcessor: PartComponentProcessorFunction<'com.enonic.app.hmdb:movie-details'> = (params) => {
+    const data = params.content.data;
+
+    // Extract the photos from content.data
+    const photos = data.photos;
+    const firstPhotoId = Array.isArray(photos) ? photos[0] : photos; // First photo ID
+    const remainingPhotoIds = Array.isArray(photos) ? photos.slice(1) : []; // Remaining photo IDs
+
+    // Fetch the first photo
+    const firstPhotoContent = getContentByKey<Content>({ key: firstPhotoId });
+    const firstPhoto = firstPhotoContent
+        ? {
+            _id: firstPhotoContent._id,
+            title: firstPhotoContent.displayName,
+            imageUrl: imageUrl({ id: firstPhotoContent._id, scale: 'width(600)' }), // Larger scale for first photo
+            id: firstPhotoContent._id
+        }
+        : null;
+
+    // Fetch remaining photos
+    const restPhotos = fetchAdditionalPhotos(remainingPhotoIds);
+
+
+    // Process the cast
+    const cast = (Array.isArray(data?.cast) ? data.cast : []).map(castMember => {
```

Comment:
Just like in other places this will return `[]` if `data.cast` is a single object.
On top of that, define a util function and import it when you need it instead of writing it everywhere.